sonic_platform_base/fan_base.py

```diff
@@ -0,0 +1,100 @@
+#
+# fan_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a fan module in SONiC
+#
+
+try:
+    import abc
+    from . import device_base
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+
+class FanBase(device_base.DeviceBase):
+    """
+    Abstract base class for interfacing with a fan module
+    """
+    __metaclass__ = abc.ABCMeta
```

Comment:
metaclasses are inherited, see https://gist.github.com/Staphylo/5597a17d781afcb7c52b3c78bbef8c32

Reply:
Thanks. Will address in next commit.